src/proto/Avalanche.proto

```diff
@@ -0,0 +1,198 @@
+    uint64 locktime = 3;          // 08 bytes
+    uint32 threshold = 4;         // 04 bytes
+    repeated bytes addresses = 5; // 04 bytes + 20 bytes * len(addresses)
+}
+
+message TransactionOutput {
+    oneof output {
+        SECP256K1TransferOutput secp_transfer_output = 1;
+        SECP256K1MintOutput secp_mint_output = 2;
+        NFTTransferOutput nft_transfer_output = 3;
+        NFTMintOutput nft_mint_output = 4;
+    }
+}
+
+message SECP256K1TransferInput {
+    uint32 typeID = 1;                   // 04 bytes
+    uint64 amount = 2;                   // 08 bytes
+    repeated uint32 address_indices = 3; // 04 bytes + 04 bytes * len(address_indices)
+}
+
+message TransactionInput {
+    oneof input {
+        SECP256K1TransferInput secp_transfer_input = 1;
+    }
+}
+
+message SECP256K1MintOperation {
+    uint32 typeID = 1;                   // 4 bytes
+    repeated uint32 address_indices = 2; // 04 bytes + 04 bytes * len(address_indices)
+    SECP256K1MintOutput mint_output = 3;          // size(mint_output)
+    SECP256K1TransferOutput transfer_output = 4;  // size(transfer_output)
+}
+
+message NftMintOpOutput {
+    uint64 locktime = 1;
+    uint32 threshold = 2;
+    repeated bytes addresses = 3;
+}
+
+message NFTMintOp {
+    uint32 typeID = 1;                   // 04 bytes
+    repeated uint32 address_indices = 2; // 04 bytes + 04 bytes * len(address_indices)
+    uint32 group_id = 3;                 // 04 bytes
+    bytes payload = 4;                   // 04 bytes + len(payload)
+    repeated NftMintOpOutput outputs = 5;          // 04 bytes + size(outputs)
+}
+
+message NFTTransferOp {
+    uint32 typeID = 1;                   // 04 bytes
+    repeated uint32 address_indices = 2; // 04 bytes + 04 bytes * len(address_indices)
+    uint32 group_id = 3;                 // 04 bytes
+    bytes payload = 4;                   // 04 bytes + len(payload)
+    uint64 locktime = 5;                 // 08 bytes
+    uint32 threshold = 6;                // 04 bytes
+    repeated bytes addresses = 7;        // 04 bytes + 20 bytes * len(addresses)
+}
+
+message TransferOp {
+    oneof operation {
+        SECP256K1MintOperation secp_mint_op = 1;
+        NFTMintOp nft_mint_op = 2;
+        NFTTransferOp nft_transfer_op = 3;
+    }
+}
+
+message UTXOID {
+    bytes tx_id = 1;       // 32 bytes
+    uint32 utxo_index = 2; // 04 bytes
+}
+
+message TransferableOp {
+    bytes asset_id = 1;           // 32 bytes
+    repeated UTXOID utxo_ids = 2; // 4 + 36 * len(utxo_ids) bytes
+    TransferOp transfer_op = 3;   // size(transfer_op)
+}
+
+message TransferableOutput {
+    bytes asset_id = 1; // 32 bytes
+    TransactionOutput output = 2;  // size(output)
+}
+
+message TransferableInput {
+    bytes tx_id = 1;       // 32 bytes
+    uint32 utxo_index = 2; // 04 bytes
+    bytes asset_id = 3;    // 32 bytes
+    TransactionInput input = 4;       // size(input)
+    repeated bytes spendable_addresses = 5; // not encoded. represented as pubkeys.
```

Comment:
should this be a string?

Reply:
not sure if this one needs to be an array, will change to string

Reply:
this is actually, public key, will rename it